ui/components-react/src/components-react/propertygrid/internal/flat-properties/FlatPropertyRenderer.tsx

```diff
@@ -98,42 +98,43 @@ export const FlatPropertyRenderer: React.FC<FlatPropertyRendererProps> = (
   const rendererManager =
     propertyValueRendererManager ?? PropertyValueRendererManager.defaultManager;
   const property = passthroughProps.propertyRecord.property;
-  const propertyRenderer =
-    property.renderer &&
-    rendererManager?.getRegisteredRenderer(property.renderer.name);
-  const typeRenderer = rendererManager?.getRegisteredRenderer(
-    property.typename
+  const customRenderer =
+    (property.renderer &&
+      rendererManager?.getRegisteredRenderer(property.renderer.name)) ||
+    rendererManager?.getRegisteredRenderer(property.typename);
+  const hasValidCustomRenderer = !!(
+    customRenderer && customRenderer.canRender(passthroughProps.propertyRecord)
```

Comment:
Maybe `rendererManager` could have method that checks if any custom renderer applies based on property record? Something like `rendererManager.hasCustomRenderer(propertyRecord)`?